android/src/main/kotlin/com/onegini/mobile/sdk/flutter/PigeonInterface.kt

```diff
@@ -200,42 +198,47 @@ open class PigeonInterface : UserClientApi, ResourceMethodApi {
     deregisterUserUseCase(profileId, callback)
   }
 
-  override fun getRegisteredAuthenticators(profileId: String, callback: (Result<List<OWAuthenticator>>) -> Unit) {
-    callback(getRegisteredAuthenticatorsUseCase(profileId))
+  override fun getAuthenticatedUserProfile(callback: (Result<OWUserProfile>) -> Unit) {
+    callback(getAuthenticatedUserProfileUseCase())
   }
 
-  override fun getAllAuthenticators(profileId: String, callback: (Result<List<OWAuthenticator>>) -> Unit) {
-    callback(getAllAuthenticatorsUseCase(profileId))
+  override fun authenticateUser(
+    profileId: String,
+    authenticatorType: OWAuthenticatorType,
+    callback: (Result<OWRegistrationResponse>) -> Unit
+  ) {
+    authenticateUserUseCase(profileId, authenticatorType, callback)
   }
 
-  override fun getAuthenticatedUserProfile(callback: (Result<OWUserProfile>) -> Unit) {
-    callback(getAuthenticatedUserProfileUseCase())
+  override fun authenticateUserPreferred(profileId: String, callback: (Result<OWRegistrationResponse>) -> Unit) {
+    authenticateUserUseCase(profileId, null, callback)
   }
 
-  override fun authenticateUser(profileId: String, registeredAuthenticatorId: String?, callback: (Result<OWRegistrationResponse>) -> Unit) {
-    authenticateUserUseCase(profileId, registeredAuthenticatorId, callback)
+  override fun getBiometricAuthenticator(profileId: String, callback: (Result<OWAuthenticator>) -> Unit) {
+    getBiometricAuthenticatorUseCase(profileId, callback)
   }
 
-  override fun getNotRegisteredAuthenticators(profileId: String, callback: (Result<List<OWAuthenticator>>) -> Unit) {
-    callback(getNotRegisteredAuthenticatorsUseCase(profileId))
+  override fun getPreferredAuthenticator(profileId: String, callback: (Result<OWAuthenticator>) -> Unit) {
+    getPreferredAuthenticatorUseCase(profileId, callback)
   }
 
-  override fun changePin(callback: (Result<Unit>) -> Unit) {
-    changePinUseCase(callback)
+  override fun setPreferredAuthenticator(authenticatorType: OWAuthenticatorType, callback: (Result<Unit>) -> Unit) {
+    callback(setPreferredAuthenticatorUseCase(authenticatorType))
   }
 
-  override fun setPreferredAuthenticator(authenticatorId: String, callback: (Result<Unit>) -> Unit) {
-    callback(setPreferredAuthenticatorUseCase(authenticatorId))
+  override fun deregisterBiometricAuthenticator(callback: (Result<Unit>) -> Unit) {
+    deregisterBiometricAuthenticatorUseCase(callback)
```

Comment:
cant we return Result from this use case?

Reply:
no, we can't beccause it's async.